cmake/modules/OmrDetectSystemInformation.cmake

```diff
@@ -193,6 +193,17 @@ macro(omr_detect_system_information)
 			if("${CMAKE_C_SIMULATE_ID}" STREQUAL "MSVC" OR "${CMAKE_CXX_SIMULATE_ID}" STREQUAL "MSVC")
 				# clang on Windows mimics MSVC
 				set(_OMR_TOOLCONFIG "msvc")
+			elseif(CMAKE_C_COMPILER_ID MATCHES "^Clang$")
+				# OpenXL17 uses CMAKE_C_COMPILER_ID "Clang"
+				set(_OMR_TOOLCONFIG "gnu")
+				set(CMAKE_C_COMPILER_IS_OPENXL TRUE CACHE BOOL "OpenXL is the C compiler")
+				if(CMAKE_C_COMPILER_IS_OPENXL)
+					set(OMR_ENV_OPENXL 1)
+					set(ENV{OMR_ENV_OPENXL} ${OMR_ENV_OPENXL})
+				else()
+					set(OMR_ENV_OPENXL 0)
+					set(ENV{OMR_ENV_OPENXL} ${OMR_ENV_OPENXL})
+				endif()
```

Comment:
This is because of my ignorance in CMake, but why have an if/else here if `CMAKE_C_COMPILER_IS_OPENXL` is set right before it? Why not just have
```
set(CMAKE_C_COMPILER_IS_OPENXL TRUE CACHE BOOL "OpenXL is the C compiler")
set(OMR_ENV_OPENXL 1)
set(ENV{OMR_ENV_OPENXL} ${OMR_ENV_OPENXL})
```
?

Reply:
I think `CACHE` variables are only written if not already set (or `FORCE`d).

Reply:
I guess this catches the case where `CMAKE_C_COMPILER_ID MATCHES "^Clang$"` but the compiler is not OpenXL? In which case this section is banking on the fact that a if using a non OpenXL compiler, some other section explicitly sets `CMAKE_C_COMPILER_IS_OPENXL ` to `FALSE` somewhere prior to this point.

Reply:
I don't see `CMAKE_C_COMPILER_IS_OPENXL` being set to false anywhere. However, the way that it's set here seems to mirror the way that `CMAKE_C_COMPILER_IS_XLCLANG` is handled further down, when using xlC as the C/C++ compiler:
https://github.com/eclipse-omr/omr/blob/d5cd947c84e2e3f56b4859f5f05497c217b506df/cmake/modules/OmrDetectSystemInformation.cmake#L216-L222
And I don't see `CMAKE_C_COMPILER_IS_XLCLANG` being explicitly set to `FALSE` anywhere either. So I'm not quite sure what exactly the reasoning is for making these `CACHE` variables.